keywords.robot

```diff
@@ -82,23 +104,23 @@ iPXE boot entry
 GRUB get menu position
     [Documentation]    Evaluate and return relative menu entry position
     ...                described in the argument.
-    [Arguments]    ${entry}
+    [Arguments]    ${entry}    ${reference_str}    ${rs_offset}
     Sleep    5s
     ${output}=    Telnet.Read
     Log    ${output}
     # enumerate output buffer lines and find line number which contain string
-    ${gnu_header}=    Get Line Number Containing String    ${output}    GNU GRUB
+    ${gnu_header}=    Get Line Number Containing String    ${output}    ${reference_str}
     ${entry_line}=    Get Line Number Containing String    ${output}    ${entry}
-    ${rel_pos}=    Evaluate    ${entry_line} - ${gnu_header} - 3
+    ${rel_pos}=    Evaluate    ${entry_line} - ${gnu_header} - ${rs_offset}
     Log    ${rel_pos}
     [Return]    ${rel_pos}
 
 GRUB boot entry
     [Documentation]    Enter specified in argument iPXE menu entry.
-    [Arguments]    ${menu_entry}
-    Set Timeout    30s
-    ${move}=    GRUB get menu position    ${menu_entry}
-    : FOR    ${INDEX}    IN RANGE    0    ${move}
+    [Arguments]    ${menu_entry}    ${reference_str}    ${rs_offset}
+    ${move}=    GRUB get menu position    ${menu_entry}    ${reference_str}    ${rs_offset}
+    ${grub_key}=    Set Variable If    ${move} < 0    ${grub_key_up}    ${grub_key}
```

Comment:
@pkonkol Please add some documentation on what is happening here.

Reply:
Added